File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        climbStairsSteps = [0, 1, 2]
        if n < 3:
            return climbStairsSteps[n]
        for i in range(3, n + 1):
            climbStairsSteps.append(climbStairsSteps[i - 1] + climbStairsSteps[i - 2])
        return climbStairsSteps[n]

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
const climbStairsSteps = [0, 1, 2]
const climbStairs = function(n) {
    if (climbStairsSteps[n]) {
        return climbStairsSteps[n];
    }
    else {
        for (let i = 3; i <= n; i++) {
            climbStairsSteps[i] = climbStairsSteps[i-1] + climbStairsSteps[i-2];
        }
        return climbStairsSteps[n];
    }
};